Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -231,7 +232,56 @@ namespace ROS2
         entity->Deactivate();
     }
 
-    static void OverrideMaterialPbrSettings(const sdf::Material* material, [[maybe_unused]] AZ::Render::MaterialAssignmentMap& overrides)
+    static void OverrideScriptMaterial(const sdf::Material* material, [[maybe_unused]] AZ::Render::MaterialAssignmentMap& overrides)
+    {
+        AZStd::string materialName(material->ScriptName().c_str(), material->ScriptName().size());
+        if (materialName.empty())
+        {
+            return;
+        }
+
+        // Make sure the material name is lowercased before checking the path in the Asset Cache
+        AZStd::to_lower(materialName);
+        // If a material has a <script> element we'll treat the name as a path and name to an O3DE material.
+        // For example, "Gazebo/Wood" will look for a product material in "<cache>/gazebo/wood.azmaterial"
+        AZ::IO::Path materialProductPath(materialName);
+        materialProductPath.ReplaceExtension(".azmaterial");
+
+        // Try getting an asset ID for the given name.
+        constexpr bool AutoGenerateId = false;
+        AZ::Data::AssetId assetId;
+        AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+            assetId,
+            &AZ::Data::AssetCatalogRequestBus::Events::GetAssetIdByPath,
+            materialProductPath.String().c_str(),
+            azrtti_typeid<AZ::RPI::MaterialAsset>(),
+            AutoGenerateId);
+
+        // No asset was found, we can't convert the material script.
+        if (!assetId.IsValid())
+        {
+            AZ_Warning("AddMaterial", false, "Failed to find product material for %s", materialProductPath.String().c_str());
```

Comment:
Calling `.String()` is not needed, since the `AZ::IO::Path` contains an `AZStd::string` internally and therefore doesn't need to make a  copy
```suggestion
            AZ_Warning("AddMaterial", false, "Failed to find product material for %s", materialProductPath.c_str());
```